SHA256SUMS fixes

NekkoDroid opened the pull request:
1. Move file generating after `mkosi.postoutput`
2. Skip directories in staging (user may have created some)
3. Remove check for directory output type (user may have added non-dir output that should be hashed)

I am a bit uncertain if I should check if there are any files to hash to skip generating the file but if the user did request creating the file they might unconditionally rely on it existing.

septatrix commented:
Intuitively I would expect to be able to access the sha sums in the postoutput script to e.g. upload them and the image to an s3 bucket for sysupdate.

DaanDeMeyer commented:
> Intuitively I would expect to be able to access the sha sums in the postoutput script to e.g. upload them and the image to an s3 bucket for sysupdate.

Indeed, I don't think we can just change this like this. The second commit looks good though

NekkoDroid commented:
> Intuitively I would expect to be able to access the sha sums in the postoutput script to e.g. upload them and the image to an s3 bucket for sysupdate.

I guess that does make sense. My thought process was more that since the files can still be created, removed and modified at that point it does make sense yet to create the file.

DaanDeMeyer commented:
> For uploading, wouldn't it make more sense to do that after everything has been moved to `OutputDirectory=`? or am I missing something

They're might be an appeal to having mkosi do the upload so one just has to run mkosi and not have to wrap it with something else. I used to recommend a wrapper script but now that we have postoutput scripts it's possible to have mkosi do it itself via a postoutput script.
